ta/arch/arm/link.mk

```diff
@@ -88,12 +88,14 @@ ifeq ($(CFG_ENCRYPT_TA),y)
 crypt-args$(user-ta-uuid) := --enc-key $(TA_ENC_KEY)
 cmd-echo$(user-ta-uuid) := SIGNENC
 endif
+user-ta-version := $(if $(CFG_TA_VERSION),$(CFG_TA_VERSION),0)
 $(link-out-dir$(sm))/$(user-ta-uuid).ta: \
 			$(link-out-dir$(sm))/$(user-ta-uuid).stripped.elf \
 			$(TA_SIGN_KEY)
 	@$(cmd-echo-silent) '  $$(cmd-echo$(user-ta-uuid)) $$@'
 	$(q)$(SIGN_ENC) --key $(TA_SIGN_KEY) $$(crypt-args$(user-ta-uuid)) \
-		--uuid $(user-ta-uuid) --in $$< --out $$@
+		--uuid $(user-ta-uuid) --ta-version $$(user-ta-version) \
```

Comment:
`$(user-ta-version)` should be enough

Reply:
Initially I tried with `$(user-ta-version)` only but build failed. Probably for computation of `if` in following statement, we need extra `$`.

`user-ta-version := $(if $(CFG_TA_VERSION),$(CFG_TA_VERSION),0)`

Reply:
OK

Reply:
Ah, I get it. The double dollar is needed because we are inside a macro definition (`gen-link-t`). With a single dollar, the `user-ta-version` is evaluated during `$(call gen-link-t)`, but at this point it is not defined. Only after `eval` is applied, the line `user-ta-verstion := $(if ...)` as some effect.

I think a better approach would be to move `user-ta-version := $(if ...)` out of the macro. In fact, close to the definition of `user-ta-uuid` in `ta/mk/ta_dev_kit.mk` would be the best place IMO.

Reply:
Makes sense. Will change accordingly.